pkg/state/state.go

```diff
@@ -585,8 +583,7 @@ func (s *stateManager) addGenesisBlock() error {
 	if _, err := s.stor.putStateHash(nil, 1, s.genesis.BlockID()); err != nil {
 		return err
 	}
-	verifyError := <-chans.errChan
-	if verifyError != nil {
+	if verifyError := chans.closeAndWait(); verifyError != nil {
```

Comment:
I think there is an incorrect order of operations. Why for apply and store diff and state hashes if `verifyError` happened? May be move `closeAndWait` up to where `close` was?